### DISCUSSION TITLE
parsed HTML nodes in syntax tree?

### DISCUSSION THREAD

**Discussion by gvwilson** [executablebooks/meta#621] [Q&A]
I am parsing documents that use `div` and `span` tags to style the Markdown. My test program is:

```
from markdown_it import MarkdownIt
from markdown_it.tree import SyntaxTreeNode

text = """\
<div class="blue">

paragraph

</div>
"""

md = MarkdownIt("commonmark")
tokens = md.parse(text)
root = SyntaxTreeNode(tokens)
print(root.pretty(indent=2, show_text=True))
div = root.children[0]
print(div.token)
```

and its output is:

```
Token(type='html_block', tag='', nesting=0, attrs={}, map=[0, 1], level=0, children=None, \
content='<div class="blue">\n', markup='', info='', meta={}, block=True, hidden=False)
```

It looks like the div is captured as a string, rather than being parsed to create nodes in the syntax tree. Is there some option I can set that'll tell MarkdownIt to do the latter, so that I get a token in the tree with tag `div` and attrs `{"class": "blue"}`?

**Comment by chrisjsewell** [comment]
Heya, so you are talking about Markdown inside HTML.

As discussed in e.g. https://stackoverflow.com/questions/29368902/how-can-i-wrap-my-markdown-in-an-html-div,
it is not explicitly supported by commonmark,
but in your example, you don't show the output of pretty: which would be an html block of the opening div, followed by parsed markdown, followed by an html block of the closing tag.
So you will end up with output html that includes the tags.
(Note there is also http://markdown-it.github.io/, to dynamicically play around with the conversion)
